lib/index.js

```diff
@@ -360,104 +398,38 @@ internals.diffTables = (migrationGroup, joinTables, allTables, mode, cb) => {
         }
     };
 
-    /*
-        Grabbed this func from https://github.com/steven-ferguson/knex-cleaner/blob/master/lib/knex_tables.js
-        `knex-cleaner` project :+1:
-
-        function getTablesNameSql(knex) {
-          var client = knex.client.dialect;
-          var databaseName = knex.client.databaseName ||
-          knex.client.connectionSettings.database;
-
-          switch(client) {
-            case 'mysql':
-              return "SELECT TABLE_NAME FROM information_schema.tables " +
-              "WHERE TABLE_SCHEMA = '" + databaseName + "' " +
-              "AND TABLE_TYPE = 'BASE TABLE'";
-            case 'postgresql':
-              return "SELECT tablename FROM pg_catalog.pg_tables" +
-              " WHERE schemaname='public';";
-            case 'sqlite3':
-              return "SELECT name FROM sqlite_master WHERE type='table';";
-            default:
-              throw new Error('Could not get the sql to select table names from client: ' +
-              client);
-          }
-        }
-
-        function getTableNames(knex, options) {
-          options = _.defaults(typeof options !== 'undefined' ? options : {}, DefaultOptions);
-
-          return knex.raw(getTablesNameSql(knex))
-            .then(function(resp) {
-              return getSqlRows(knex, resp)
-                .map(function(table) {
-                  return table[Object.keys(table)[0]];
-                })
-                .filter(function(tableName) {
-                  return !_.contains(options.ignoreTables, tableName);
-                });
-            });
-        }
-    */
-
-    // TODO get a list of the tables in the knex here, and compare against
-    // allTables to see if tables in the DB need to be deleted.
-
-    // Use something similar to this block that I accidentally made for columns
-    // D'oh!
-    /*
-        Promise.all(allDrops.map((tableName) => {
-
-            return new Promise((resolve, reject) => {
-
-                internals.getColumnsForTable(knex, tableName)
-                .then((dbSchema) => {
-
-                    modelDeltaObject.drop.push({
-                        tableName: tableName,
-                        columns: dbSchema
-                    });
-                    resolve();
-                })
-                .catch((err) => {
-
-                    reject(err);
-                })
-            });
-        }))
-        .then(() => {
-
-            knexTableDeltaObject.tableDelta.drop.push(modelDeltaObject);
-            next();
-        });
-    */
-
     Items.serial(Object.keys(allTables), (tableName, tableNext) => {
 
         // Ping the db for existence of each table
 
-        knex(tableName).select(knex.raw('1')).asCallback((err) => {
+        Promise.resolve(knex(tableName).select(knex.raw('1')))
```

Comment:
I would have expected `knex(tableName).select(knex.raw('1'))` to have an `asCallback()` method without wrapping it.  Not the case?